app/controllers/application_controller.rb

```diff
@@ -1104,10 +1104,10 @@ def get_view_process_search_text(view)
                "%#{stxt}%"
              end
 
-      id = @search_text if /^\d+$/.match?(@search_text)
+      id = @search_text.to_i if /^\d+$/.match?(@search_text)
       condition = [[]]
-      if id
-        add_to_search_condition(condition, "#{view.db_class.table_name}.id = ?", id.to_i)
+      if id > 0 && id <= 9223372036854775807
```

Comment:
We should set id to an invalid value like -1, so that in the end it return no records.  What this will do is ignore the input from the user, and return _all_ the records.

Reply:
if they pass in a string "abc", then the search will use a `like '%?%'`
If they pass in a string "123" (small), then we use `like` and `id = ?`
If they pass in a string "123" (really big), then we will use a `like`.

So whether this is a string or a really big integer, they both seem like they will behave the same way.

I will see if I can throw a test in here

Reply:
Also, this code will probably fail as is because it compares a nil with an integer when there incoming search text is regular text.

Reply:
This is what I was thinking:

```
      id = @search_text.to_i if /^\d+$/.match?(@search_text)
      id = -1 if id && id > 9223372036854775807
      condition = [[]]
      if id
        add_to_search_condition(condition, "#{view.db_class.table_name}.id = ?", id)
      end
```

Reply:
Yes that's what's annoying about this field, its a single string field but changes how it queries depending on it's incoming shape.

Reply:
Oh wait .. In the middle case is it `like and id` or `like or id`?

Reply:
suggest that `9223372036854775807` be put into a const with a meaningful name

Reply:
We were chatting offline and I was suggesting a patch to Rails itself to expose that as a constant (since they have the magic number too)